tictactoe/src/App.js

```diff
@@ -80,34 +119,120 @@ export default function Game() {
     );
   });
 
+    
+  const status = gameStatus(currentSquares, xIsNext);
+
   return (
     <div className="game">
       <div className="game-board">
-        <Board xIsNext={xIsNext} squares={currentSquares} onPlay={handlePlay} />
+        <Board status={status} xIsNext={xIsNext} squares={currentSquares} onPlay={playerMove}  playWithBot={playWithBot}/>
       </div>
       <div className="game-info">
+        <div class="checkbox-container">
+        <Checkbox
+        label="Play against bot"
+        value={playWithBot}
+        onChange={() => setPlayWithBot(p => !p)}
+        />
+         <Checkbox
+        label="Enter playernames"
+        value={!playWithBot}
+        onChange={() => setPlayWithBot(p => !p)}
+        />
+</div>
         <ol>{moves}</ol>
       </div>
     </div>
   );
 }
+const lines = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [0, 4, 8],
+  [2, 4, 6]
+];
 
 function calculateWinner(squares) {
-  const lines = [
-    [0, 1, 2],
-    [3, 4, 5],
-    [6, 7, 8],
-    [0, 3, 6],
-    [1, 4, 7],
-    [2, 5, 8],
-    [0, 4, 8],
-    [2, 4, 6]
-  ];
-  for (let i = 0; i < lines.length; i++) {
-    const [a, b, c] = lines[i];
-    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
-      return squares[a];
+  const winningLines = matchingLines(squares, line => {
+    return line[0] && line[1] === line[0] && line[2] === line[0];
+  });
+
+  if (winningLines.length > 0)
+    return squares[winningLines[0][0]];
+  return null;
+}
+
+function gameStatus(squares, xIsNext) {
+  const winner = calculateWinner(squares);
+  let status;
+  if (winner) {
+    status = "Winner: " + winner;
+  } else {
+    status = "Next player: " + (xIsNext ? "X" : "O");
+  }
+  return status;
+}
+
+function moveLeft(squares) {
```

Comment:
The function nomenclature is a little ambiguous. After reading it the first time, I got the impression that it's to define the left movement, although it's to designate the number of remaining movements. I'd suggest naming it "restMoves()."